stable/mongodb-replicaset/values.yaml

```diff
@@ -3,8 +3,8 @@ nameOverride: ""
 fullnameOverride: ""
 
 # Override the deployment namespace
-# global:
-#   namespaceOverride: ""
+global:
+  namespaceOverride: ""
```

Comment:
@srinivasa-vasu, Uncommented global values won't allow for override from top level except when they are directly in the final values file which is against the point of global values.

That is why in most cases global values are commented, example in [redis chart](https://github.com/helm/charts/blob/a9c0951a1e27a2577c2fb2ee26bf3144880d2aef/stable/redis/values.yaml#L1-L9), also [similar checks](https://github.com/helm/charts/blob/a9c0951a1e27a2577c2fb2ee26bf3144880d2aef/stable/redis/templates/_helpers.tpl#L74-L87) has been done to ensure the global values existence 

I think a workaround would be defining a dummy value in the global section to make the global valid, but TBH, I am not sure how much we can gain from that comparing from the confusion that it may caused by moving the complexity from _helper.tpl to values.yaml file,

Is it possible for you explain a little more about the reason of this change?

Reply:
@mforutan - Got it. helm3 is not properly honoring the hasKey logic. I've reverted the global change. Instead changed the conditional logic to be compatible with both helm2 and 3.